### PR TITLE
Floor divide #17658

### LINKED ISSUE #17658: floor_divide
will implement floor_divide paddle frontend function

### PR DESCRIPTION
**Pull request by alovega**
Close #17658

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by alovega** [comment]
Ooh sorry, force of habit, I have updated the pr description

**Comment by hmahmood24** [comment]
Hey @alovega, thanks a lot for the PR. Everything looks good to me! 🚀 Can you just resolve the merge conflicts so we can merge the PR? Thanks! 😃

**Comment by alovega** [comment]
Hello @hmahmood24 the merge conflicts are fixed please review

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -195,8 +195,9 @@ def multiply(self, y, name=None):
     def isfinite(self, name=None):
         return ivy.isfinite(self._ivy_array)
 
-    def all(self, axis=None, keepdim=False, name=None):
-        return ivy.all(self.ivy_array, axis=axis, keepdims=keepdim)
+    @with_supported_dtypes({"2.5.0 and below": ("float16", "bfloat16")}, "paddle")
+    def all(self, axis=None, keepdim=False, dtype=None, name=None):
+        return ivy.all(self.ivy_array, axis=axis, keepdims=keepdim, dtype=dtype)
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/tensor.py:200]: Are changes to this function necessary for your function? If not, it'd be better to revert these changes since we follow a 1 function 1 PR policy for maintainability purposes

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -276,3 +277,11 @@ def greater_than(self, y, name=None):
     @with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")
     def rsqrt(self, name=None):
         return ivy.reciprocal(ivy.sqrt(self._ivy_array))
+
+    @with_supported_dtypes(
+        {"2.5.0 and below": ("int32", "int64")},
+        "paddle",
+    )
+    def floor_divide(self, y, name=None):
+        y_ivy_array = _to_ivy_array(y)
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/tensor.py:286]: We don't need this line here. You can remove this

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 3]
```diff
@@ -276,3 +277,11 @@ def greater_than(self, y, name=None):
...
     def rsqrt(self, name=None):
         return ivy.reciprocal(ivy.sqrt(self._ivy_array))
+
+    @with_supported_dtypes(
+        {"2.5.0 and below": ("int32", "int64")},
+        "paddle",
+    )
+    def floor_divide(self, y, name=None):
+        y_ivy_array = _to_ivy_array(y)
+        return ivy.floor_divide(self._ivy_array, y_ivy_array)
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/tensor.py:287]: Here, instead you can pass in `y._ivy_array`. The `_ivy_array` attribute inside the `paddle.Tensor` objects already contains their `ivy.Array` equivalent so we don't need to explicitly convert `y` to an ivy array

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 4]
```diff
@@ -1448,3 +1448,42 @@ def test_paddle_greater_than(
...
     )
+
+
+# floor_divide
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="floor_divide",
+    dtypes_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1459]: Best to keep this as `helpers.get_dtypes("valid")` for wider test coverage

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 5]
```diff
@@ -1448,3 +1448,42 @@ def test_paddle_greater_than(
...
+def test_paddle_floor_divide(
+    dtypes_and_x,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+):
+    input_dtype, x = dtypes_and_x
+    assume(not np.any(np.isclose(x[1], 0)))
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1476]: This doesn't need to be included here. Instead you can pass in `min_value` argument to the `helpers.dtype_and_values(...)` such that the minimum value in the generate arrays is greater than 0. We generally don't recommend using `assume` inside the testing function unless absolutely necessary

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @alovega, requested a few changes on the PR. Can you please look into these? Thanks! :)

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 1 above)
Comment (alovega) [reply ivy/functional/frontends/paddle/tensor/tensor.py:200]: I think the changes came when I merged changes from the upstream master, should I do the changes

Comment (hmahmood24) [reply ivy/functional/frontends/paddle/tensor/tensor.py:200]: Yes please :)

**Comment by alovega** [comment]
Hi @hmahmood24 I have resolved the requested changes please review and get back

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 6]
```diff
@@ -328,3 +328,7 @@ def fmin(self, y, name=None):
     def minimum(self, y, name=None):
         y_ivy = _to_ivy_array(y)
         return ivy.minimum(self._ivy_array, y_ivy)
+
+    @with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+    def floor_divide(self, y, name=None):
+        return ivy.floor_divide(self._ivy_array, y)
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/tensor.py:334]: This was supposed to be `y._ivy_array`. Maybe a mistake?

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @alovega, can you please revert the changes you pushed to the `array_api_testing` folder? Also requested one more change. Thanks :)

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 6 above)
Comment (alovega) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: I have tried to explicitly convert passed argument y to an ivy_array using the y._ivy_array attribute but on testing, I  am getting an error saying the paddle.Tensor object has no attribute ._ivy_array so I have reverted back the changes.
May

Comment (alovega) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: for more info to see how the tests are failingScreenshot 2023-06-28 at 13 59 50

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 7]
```diff
@@ -345,6 +345,11 @@ def minimum(self, y, name=None):
         y_ivy = _to_ivy_array(y)
         return ivy.minimum(self._ivy_array, y_ivy)
 
+    @with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+    def floor_divide(self, y, name=None):
+        y_ivy = _to_ivy_array(y)
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/tensor.py:350]: Apologies for the confusion. You can use something like: `y_ivy = y._ivy_array if isinstance(y, Tensor) else _to_ivy_array(y)`. I think that should resolve the issue

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @alovega, apologies for the confusion earlier. Suggested the change and also could you please revert or rebase the change to `array_api_testing` folder? It should not be a part of your PR's changed files before we merge it into `master`. Thanks a lot!

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 7 above)
Comment (alovega) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: Thank you @hmahmood24, I have done a rebase to upstream master and the requested update please review and get back.

**Review by hmahmood24 — APPROVED** [review]

**Comment by hmahmood24** [comment]
Hey @alovega, LGTM now! Thanks for making this contribution! 🚀

**Comment by alovega** [comment]
Thanks @hmahmood24
